### ISSUE TITLE
WebAIM Feedback - SC 1.4.11 – Resize content

### ISSUE THREAD

**Issue by jared-w-smith** [w3c/wcag21#250]
A primary concern is that this Level A success criterion requires 400% text sizing which would invalidate the existing 1.4.4 Level AA success criterion which requires 200% text sizing. This would cause confusion as to why a Level AA SC requires a lower threshold for sizing than a new Level A SC.

Meeting this SC will require significant effort by many web authors. Few popular sites tested come close to meeting the 400% sizing threshold.

WebAIM suggests a more practical approach – perhaps 200% text sizing with minimal horizontal scrolling (perhaps double the viewport width) at Level A, 400% text sizing with minimal horizontal scrolling (perhaps double the viewport width) at Level AA, and the draft SC as proposed at Level AAA.

It is hoped that the working group will also address content loss for true text sizing (i.e., not zooming the page, but only increasing text size) to a reasonable level – perhaps 150%. This and the existing Resize Text SC generally neglect this audience which [represents 38% of respondents to the WebAIM Survey of Users with Low Vision](http://webaim.org/projects/lowvisionsurvey/#at).

**Comment by mraccess77** [comment]
@jared-w-smith I don't see how supporting 400% neglects the audience who needs 150%.  It totally supports that audience in every way.  Regarding 150% magnification -- at what screen resolutions and other display factors are used?  There are many factors to consider with zoom.

**Comment by jared-w-smith** [comment]
There are factors when increasing only text size (not zooming the entire page) that will impact readability and functionality in the page that are not present when utilizing page zoom. With success criteria that only address and require page zoom/magnification, the needs of users who increase only text size are not fully addressed.

**Comment by WayneEDick** [comment]
Please read http://nosetothepage.org/Fitz/2dScroll.html

Nobody ever evaluated the cost of horizontal scrolling before. The overhead
is unacceptable.  It always was. It just took a mathematician with
experience using horizontal scrolling to compute the cost.

It is in light of this computation the Low Vision Task Forced resolved on 9
Feb 2017 that enlargement without word wrapping is not accessibility
support. The vote was unanimous.

As a committee, we believe there is no acceptable amount of horizontal
scrolling.

Responsive design applied to mobile phones showed that enlargement up 500%
was manageable. There is no reason to subject people with low vision with
an cruel reading interface any more.

**Comment by jared-w-smith** [comment]
@WayneEDick  -

I'm not suggesting that scrolling is not impactful. It certainly is. But the notable author impact of avoiding scrolling at 400% zoom also needs to be considered. If WCAG is rejected due to introducing overly onerous requirements, then everyone loses.

Of note is that [only 25% of low vision users indicated needing magnification to 400% or more](http://webaim.org/projects/lowvisionsurvey/#magnification). Only 42% needed larger than 200%. Compare that with [91% of respondents that found color contrast important](http://webaim.org/projects/lowvisionsurvey/#magnification), yet contrast starts at only Level AA in WCAG due (at least partially) to its impact on design.

Our recommendations simply suggest a practical consideration of both user and author interests.

**Comment by WayneEDick** [comment]
Dear Jared and Everyone,


We can answer the question of what people with low vision prefer. I am
dubious of the choice of 200% over 400% or higher. I'm thinking 200% can
occur without word wrapping but in most peoples experience 400% does not
occur in that format. As I remember the WebAIM survey occurred before
responsive web design.

We can settle that question. I have written a  program that enables choice.
The user needs no skill at AT. It just enables the user to select their
preferred format. It gives every choice in adaptive text plus any font size
imaginable.

The program is https://trx.knowbility.org/

Wayne

**Comment by DavidMacDonald** [comment]
@jared-w-smith 
I think if the linearization SC works out it would overcome text zoom (150%). The problem with 150% text zoom is that it make the CSS incredibly complex. Clipping text etc... would require a major redesign for most web sites.

**Comment by WayneEDick** [comment]
There were two problems with 1.4.4 Resize Text. It under estimated needed
font size. Note: The WebAIM survey shows only 25% of people with low vision
prefer text larger than 200%. That appears like a small group, but 25% of
the population with low vision is a population as large as all non-visual
users. Also, 1.4.4 underestimated the insufficiency of failure to word
wrap. Enlargement without word wrapping was considered accessibility
support.

In light of formal operational analysis,  zoom without word wrapping does
not provide support that is comparable to screen reader access with
refreshable braille. You cannot call a process support, if it requires 40
to 100 times the keyboard / mouse input just to perceive the content. It
also does not provide enough perceivable content at any time to support
memory while listening. I think everyone thought lack of word wrapping was
awkward, but I do not think anyone realize the seriousness of the defect.

Level A does not just measure ease of implementation. It measures
importance. For people with reduced visual acuity font size is the critical
accommodation. Without sufficient font size other assistance is moot. It
makes little sense to set 200% as a max. It takes 1/3 normal acuity to
qualify as having LV with the World Health Organization. 300% would make
more sense. The problem is the user must face a trade off between
sufficient enlargement and loss of word wrapping. It's a devil's bargain.
When faced with that choice, I choose the smaller size and read with a hand
held magnifier. That does not mean that I prefer 200%. It is just that I
will do a lot to avoid lines that run off the screen.

Assigning level AA to 1.4.4 was just a mistake. WCAG 2.1 needs to fix that
mistake.

**Comment by DavidMacDonald** [comment]
@jared-w-smith 
Would you be ok with 300%? We expect most sites would meet this by Responsive.

**Comment by jared-w-smith** [comment]
@DavidMacDonald It's really a balance between user needs and author support/acceptance. This is why we suggested a tiered approach with each level. I think 300% is more reasonable for higher levels.

FWIW, supporting 400% (or even 300%) zoom without horizontal scrolling is, I think, generally much more difficult to support than 150% text sizing. The former requires a significant site and CSS overhaul. The latter typically necessitates one thing - replacing `height` with `min-height` in CSS and sometimes being careful with absolute positioning of elements that contain text.

**Comment by DavidMacDonald** [comment]
@jared-w-smith Sounds like you have the beginning of a technique there for 150% text zooming. @allanj-uaag @WayneEDick  What you guys think about the 150% text size increase for text zoom only as either an additional requirement of an optional requirement?

**Comment by alastc** [comment]
A few things I'd like to pick up on:

> Few popular sites tested come close to meeting the 400% sizing threshold.

I disagree, many do. If you start with: A laptop/desktop browser at 1280px wide, and zoom to 400% many popular (responsive) sites meet that now.

The reason for that is simple, 1280/4 = 320px, the width of the original iPhone. Therefore any responsive design that aims to cater for 'smaller' phones (including the iPhone 5s and some androids I think) will cater to 400% zoom.

I've just checked the home pages of: BBC.co.uk, Adobe.com, microsoft.com, theguardian.com, webaim.org (barring the logo going behind the menu), all good at 400%.

I'm sure there are counter examples, but to say it is impossible or even particularly difficult is not correct. I've been over this with a few clients recently and the principle is not scary, the details are  about catching things like sticky headers which assume a portrait view, whereas non-mobiles are landscape. (Vertical media queries to un-stick the menu solve that.)

> A primary concern is that this Level A success criterion requires 400% text sizing which would invalidate the existing 1.4.4 Level AA success criterion which requires 200% text sizing. 

What to do with (or in relation to) 1.4.4 is a good question, it is useful for catching the exceptions in this SC (e.g. text within data-tables). However, we then need to balance not changing a WCAG 2.0 SC with WCAG 2.1 making sense as a whole. Would you be happy to change a 2.0 SC to adapt to a related new 2.1 SC?

> It is hoped that the working group will also address content loss for true text sizing (i.e., not zooming the page, but only increasing text size)... which represents 38% of respondents to the WebAIM Survey of Users with Low Vision.

I'm sure there are many people that do still resize text (only), a couple of years ago there was a lot of resistance to zoom & responsive design. However, I think many people have come around to the idea that you (the user) can get a lot more control and increased size when authors can adapt the layout as zoom increases. 

@WayneEDick @allanj-uaag would you agree?

I still haven't heard from anyone about a negative aspect of zoom+responsive that isn't due to a non-responsive layout. It will be a while before this change to responsive percolates through to user-surveys though, perhaps having this SC in place can help with the education process?

**Comment by alastc** [comment]
Hi @jared-w-smith,

The SC text has gone through significant revision, [latest version here](https://rawgit.com/w3c/wcag21/resize-content_ISSUE-77/guidelines/sc/21/resize-content.html).

* I have written about the [necessity & feasibility aspects](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Resize_content_issues_review#Initial_response:) and (with the exception in place) it is  feasible for any site, and many popular sites meet it already.
* It is currently at level AA, I'd like to raise the current 1.4.4 to level A to make it more understandable as a whole, but I think that's dealt with as a separate issue.

Does that mitigate your issues on this SC?

**Comment by jared-w-smith** [comment]
@alastc We originally voiced 3 concerns:

1. It conflicts and essentially overrides existing SC.
2. It may require significant effort by many web authors.
3. It does not address the needs of users who increase text size only.

#1 - We would support 1.4.4 being adjusted to Level A (if the exceptions for this SC are added) and this SC being defined at AA. Otherwise there will certainly be confusion.

#2 - I'm not sure it has fully been considered how this will impact most web sites and applications,  though the exceptions do provide some relief here.

> Overall, there are many high-traffic, high profile sites that either meet this criteria, or could meet it without significant investment thanks to changes in development techniques.

I don't believe this to be true. A quick check of the [Alexa top 50 home pages](http://www.alexa.com/topsites) shows that 4 of the 50 (8%) currently pass this SC (even considering the current exemptions). Most fail spectacularly. I wouldn't consider 8% of the most popular sites on the planet to be "many".

Ultimately the working group must consider and balance author impact with end user impact and consider how these will impact adoption of 2.1. We still believe that the author impact of this SC would be very significant.

#3 is outside the scope of this SC, though I think it warrants consideration elsewhere.

**Comment by alastc** [comment]
Ok, let's tackle that list:

- Google.com, a completely fixed design that does mobile detection. If I change the reported UA to a phone, it passes easily. Let's call this approach 'adaptive'.
- Youtube, as above, adaptive.
- Facebook, adaptive.
- Baidu, adaptive, although I'd recommend some vertical media queries to prevent the floating header from taking up too much space.
- Wikipedia, looks good to me ([starting here](https://en.m.wikipedia.org/wiki/Main_Page)).
- Yahoo, as per Baidu.
- Qq.com, not in the least bit responsive (or adaptive). It's a Chinese site, and (as per the machinations on what direction scrolling needs to account for), I'd like more info from people familiar with that aspect of internationalisation, I think there's more going on there.
- Reddit almost works at 400% (ok at 200%), but it is also adaptive when using a mobile UA.

So at the top, TOP level of sites, they have enough design/tech resource to have multiple versions of their sites rather than rely on responsive techniques. They check their stats, make sure they cover 99.x% of mobile UAs and have probably never considered that some people need to make things significantly bigger.

With sufficient motivation (e.g. a WCAG SC) perhaps they would consider adding a link to the mobile version, or detecting the zoom level and triggering the mobile layout, or even moving to a more generally responsive approach. They have the pieces in place, they just need to be available. (But with a caveat about the Chinese / non-western language sites, need more info there).

When you move down the level of sites, such as national level, I find most have already made the switch to responsive. E.g. usa.gov, gov.uk, bbc.co.uk, microsoft.com, argos.co.uk (e-commerce), nationaltrust.org.uk.

Those that haven't switched to responsive are probably planning it, I know those few of our clients that haven't are planning to. Not for accessibility reasons, but to practically support the multitude of devices out there.

From other feedback, most of the examples provided that are not responsive appear to have front-end code from before 2013(ish), before responsive caught on.

We need this in place soon so we can both encourage people to support this level of zoom, but also catch issues with responsive sites. (Sticky headers come up a lot!)

Given that Flexbox now has good support, and CSS grids have wide support (just waiting on Edge), the tools are there, but WCAG doesn't cover it.

**Comment by jared-w-smith** [comment]
@alastc Thank you for this additional detail. A few follow ups...

> Google.com, a completely fixed design that does mobile detection. If I change the reported UA to a phone, it passes easily.

I think you're just stating this to indicate that they have a framework in place for supporting this level of zoom, correct? I just want to confirm that you're not saying that end users changing their UA to some device that results in a conformant version is a sufficient technique for this SC.

> With sufficient motivation (e.g. a WCAG SC) perhaps they would consider adding a link to the mobile version

Are you saying that providing a link to a "mobile version" of a conformant site would be a sufficient technique?

I'm increasingly on board with this (at Level AA as outlined previously), but I do think there will be difficulties in getting this adopted, particularly for complex web applications and legacy content.

**Comment by alastc** [comment]
> I think you're just stating this to indicate that they have a framework in place for supporting this level of zoom, correct?

Yep, I'm not suggesting that users do that, just that it is something they already have in place that indicates it is feasible.

As an aside, the BBC (now all responsive) started by making a new 'mobile' version of their news section that was essentially responsive but only shown to mobile devices. Then they expanded that to cover desktop, and eventually stopped doing device detection. It's a neat approach several of our clients are using to switch.

> Are you saying that providing a link to a "mobile version" of a conformant site would be a sufficient technique?

Yes, in the general sense that you can link to an alternative conforming version. It should provide equivalent content & functionality though. This github site has a mobile version (if you do manage to switch UA), but it is a sub-set for features, so I wouldn't consider that conforming.

> I'm increasingly on board with this (at Level AA as outlined previously), but I do think there will be difficulties in getting this adopted, particularly for complex web applications and legacy content.

Legacy content has come up before and I think the general thought process was that WCAG 2.0 is what still applies to legacy content, and you make a conformance claim for 2.1 for new content. The 2.0 version is not retired by the W3C, it is up to people to claim conformance against a particular version, or your policy people (e.g. Government) to say what you should conform to.

In some places that's things like the Section 508 process. In the UK there isn't a specified standard, it is what a court thinks is reasonable.

For complex web-apps, we've actually had very few comments about that from people (developers) who actually have to make it work. I hesitate to point out individuals or particular companies, but I would note that Oracle have commented, and didn't raise feasibility. I suspect because they have a [library of complex widgets](http://www.oracle.com/webfolder/technetwork/jet/jetCookbook.html?component=home&demo=all) for use in web-apps that are all responsive.

There was a comment from Microsoft about things like powerpoint interfaces, which drove a couple of the exceptions.

When I talk to various people about I find managers (project/product) are worried about it, some consultants are worried, but not the people who actually implement it. Yes it can be a big change if they aren't going responsive, but they are, and not for accessibility reasons!

It is at AA, have I swayed you yet? :-)

**Comment by jared-w-smith** [comment]
>> Are you saying that providing a link to a "mobile version" of a conformant site would be a sufficient technique?

>Yes, in the general sense that you can link to an alternative conforming version. It should provide equivalent content & functionality though.

Hmm... I'm not sure about this. Is it reasonable to expect low vision users to know to click "mobile version" to get a version that works for them? How will the user reasonable find this link on a site that does not support zoom? (Yes, this is a concern about conforming versions in general, though this one seems particularly acute.)

> It is at AA, have I swayed you yet? :-)

Mostly. :-)

**Comment by alastc** [comment]
> Is it reasonable to expect low vision users to know to click "mobile version" to get a version that works for them?

Hmm, thinking.

There are a few current and future possibilities:

- Search for 'mobile' in text on the site (hacky!)
- The site could detect the zoom level (or rather, the desktop UA + apparent viewport size) and show that link prominently.
- Use a (currently non-existant) user preference. There's work on preferences going on elsewhere it could hook into.
- Programmatically associate that link with something, although I'm not sure what yet.

I think the first two are feasible in the 2.1 timeline, not ideal obviously, but they are likely to be temporary and still better than the current situation.

**Comment by alastc** [comment]
Hi @jared-w-smith, have you seen the [latest text for zoom content](https://w3c.github.io/wcag21/guidelines/#zoom-content)?

How are we doing from your point of view? (Of note, a comment from TPG: #335).

**Comment by jared-w-smith** [comment]
Many of our original concerns have seemingly not been addressed - there's no consideration in WCAG for text-only zoom and no resolution (that I see) regarding this SC essentially invalidating two other existing SC. But those are outside the scope of the SC language itself.

It's clear consideration has gone into the changes to this SC and we're very supportive of it. TPG raised some valid concerns - and you had a clear explanation of the changes. I'll be curious to see if/how those are resolved.

**Comment by steverep** [comment]
> there's no consideration in WCAG for text-only zoom

Explicitly that's true, but the Adapting Text SC requires authors to accommodate expansion in character, word, and line spacing.  For a given piece of text, this could also be mapped to a font size increase.  I suggest we discuss text-only zoom in the context of that SC and not this one since the author responsibility is really about container sizing, whereas Zoom Content is much more about the layout of those containers.

**Comment by alastc** [comment]
Wait, two others? I can see "resize text" is less important (but still needed for small-screen devices where the layout cannot adapt), which other one?

To build on Steve's comment, we've been testing the 'adapting text' SC on the top 50 sites, and it is pretty good for finding containers that have a fixed height. It seems like a reasonable ask, less than the original 200% text-size, but does allow for some buffer.

Combined with 400% zoom, and you've got a pretty good solution. In fact, maybe that does mean resize text is un-needed now... damn I'm talking myself around.

**Comment by jared-w-smith** [comment]
> Wait, two others? I can see "resize text" is less important (but still needed for small-screen devices where the layout cannot adapt), which other one?

You're correct. For some reason I was thinking that there was a AAA sizing SC. It's just 1.4.4 Resize Text that would essentially be overridden by this SC.

I'll look further into the adapting text SC - it seems that's likely a better approach anyway (which, if so, would further invalidate 1.4.4).

Sorry for the confusion.

**Comment by alastc** [comment]
Hi everyone, I think we worked through most issues in this thread, I'd like to close _this_ issue.

I will note that there are other issues/discussion in github that continue, broadly there are two themes:
- Feasibility (similar to this one)
- Internationalisation.

So although I'm closing this one, it is far from over! If you think there is more to go through here, please do re-open the issue.
